Doc/library/pathlib.rst

```diff
@@ -373,6 +372,9 @@ Pure paths provide the following methods and properties:
    .. versionchanged:: 3.10
       The parents sequence now supports :term:`slices <slice>` and negative index values.
 
+   .. versionchanged:: 3.12
+      Type changed from a tuple-like immutable sequence to a true tuple.
+
```

Comment:
FYI @AlexWaygood, this is probably the most questionable part of the patch. Would be grand to get your input on whether it's reasonable or not! Thanks

Reply:
Hmm, I'm pretty uncomfortable about doing this without a deprecation warning, I'm afraid :( It feels like quite a big backwards-compatibility break, given that you can do this on `main`:

```pycon
>>> from pathlib import Path
>>> x = Path("foo", "bar", "baz").parents
>>> x
<WindowsPath.parents>
>>> x[0]
WindowsPath('foo/bar')
>>> x[1]
WindowsPath('foo')
```

Reply:
With this PR:

```python
>>> from pathlib import Path
>>> x = Path("foo", "bar", "baz").parents
>>> x
(PosixPath('foo/bar'), PosixPath('foo'), PosixPath('.'))
>>> x[0]
PosixPath('foo/bar')
>>> x[1]
PosixPath('foo')
```

Reply:
Yep, you're quite right! I think I mistyped something in the REPL, and confused myself :)

Many apologies!